lib/lists.nix

```diff
@@ -200,6 +200,25 @@ rec {
     let found = filter pred list;
     in if found == [] then default else head found;
 
+  /* Takes a predicate and a list as input and
+     returns a list of lists, separated by elements that match the predicate.
+
+     This is analoguous to `builtins.split separator string`,
+     with a predicate instead of a separator
+     and a list instead of a string.
+
+     Type: splitList :: (a -> bool) -> [a] -> [ (a | [a]) ]
+
+     Example:
+       splitList (x: x == "x") [ "y" "x" "z" "t" ]
+       => [ [ "y" ] "x" [ "z" "t" ] ]
+  */
```

Comment:
can you document what happens when no element or several contiguous elements satisfy the predicate ?

Reply:
OK, I documented this. Please tell me if that suits you.

Reply:
The examples explain that very well, thanks.